File: app/src/main/java/app/android/ttp/mikazuki/yoshinani/binding/BindableString.java
package app.android.ttp.mikazuki.yoshinani.binding;

import android.databinding.BaseObservable;
import android.support.annotation.Nullable;

import org.parceler.Parcel;

import java.util.Objects;

/**
 * @author haijimakazuki
 */
@Parcel
public class BindableString extends BaseObservable {

    String mValue;

    public String get() {
        return mValue != null ? mValue : "";
    }

    public void set(@Nullable final String value) {
        if (!Objects.equals(mValue, value)) {
            mValue = value;
            notifyChange();
        }
    }

    public boolean isNotEmpty() {
        return !isEmpty();
    }

    public boolean isEmpty() {
        return mValue == null || mValue.isEmpty();
    }
}

File: app/src/main/java/app/android/ttp/mikazuki/yoshinani/binding/BindableString.kt
package app.android.ttp.mikazuki.yoshinani.binding

import android.databinding.BaseObservable

import org.parceler.Parcel

import java.util.Objects

/**
 * @author haijimakazuki
 */
@Parcel
class BindableString : BaseObservable() {

    internal var mValue: String? = null

    val isNotEmpty: Boolean
        get() = !isEmpty

    val isEmpty: Boolean
        get() = mValue == null || mValue!!.isEmpty()

    fun get(): String {
        return if (mValue != null) mValue else ""
    }

    fun set(value: String?) {
        if (mValue != value) {
            mValue = value
            notifyChange()
        }
    }
}